src/shipit_static_analysis/tests/test_clang.py

```diff
@@ -85,3 +86,15 @@ def test_clang_format(tmpdir):
 
     # At the end of the process, original file is patched
     assert bad_file.read() == BAD_CPP_VALID
+
+    # Test stats
+    stats, reporter = mock_stats
+    stats.api.flush(time.time() + 20)
```

Comment:
This is a bit flaky 😄 
Perhaps we can add a function to the mock to wait for the flushing, but not sure it's worth the effort.